### DISCUSSION TITLE
Help user know which key input is being Configured in the Event Configuration dialog

### DISCUSSION THREAD

**Discussion by nrichards** [godotengine/godot-proposals#8682] [Editor]
Currently when editing the Project Settings > Input to add and configure events, the configuration dialog is ambiguous about which event is being configured. It is moderately easy for me to mis-configure physical keys to events with an errant mouse click.

Proposal: add the event name to the event configuration dialog, in parentheses. E.g. "Event Configuration (move_forward)" instead of the current "Event Configuration".

Scenario

1. I want to configure the "W" key for my "move_forward" input event
2. I open the Project Settings > Input, and see "move_forward". I already have that, "move_backward", "move_left", and "move_right". 
3. I select the "+" button at the far right side of the Input tab, for what I think is the "move_forward" event.
4. A dialog pops up titled, "Event Configuration", which listens for my input. I type "W", and click OK.
5. I see the "W (physical)" key added to my input map.
6. ISSUE: I clicked the "move_left" button, and "W" is **not** attached to "move_forward".

https://github.com/godotengine/godot-proposals/assets/704698/91faf349-a423-4edd-9d1c-e598aa6ff62f

I wonder if it is in the style of the Godot Editor UI to add a clarifying piece of text to pop up dialogs? I see this pattern in the Project Settings dialog itself, which includes the name of the project file in parentheses. Is this a "bad" UI style, overall, or may we build on top of that?

**Comment by Calinou** [comment]
I tried my hand at implementing this, but it only works when editing existing events in an action, not when adding a new event in an action: https://github.com/Calinou/godot/tree/editor-input-map-editor-action-edit-title

I have no idea why :slightly_frowning_face:

**Reply by nrichards** [reply]
I traced the code and, on my system the `set_title` calls made after the `InputEventConfigurationDialog()` ctor are not reached. The default title is used instead because `if (p_event.is_valid()) {` fails in `::popup_and_configure`.

**Reply by nrichards** [reply]
[The WIP commit](https://github.com/godotengine/godot/commit/95853006c582481c8c6131553f234271eece1298) works for Edit Action. 

For Add it fails since the `input_event` is empty. In that empty case, to make Add work there could be some more `set_title`s in the `else` of `::popup_and_configure`:

```
	} else {
		// Clear Event
		_set_event(Ref<InputEvent>(), Ref<InputEvent>());

                // ADD THIS STUFF vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
		if (!p_current_action_name.is_empty()) {
			set_title(vformat(TTR("Event Configuration for \"%s\""), p_current_action_name));
		} else {
			set_title(TTR("Event Configuration"));
		}

		// Clear Checkbox Values
		for (int i = 0; i < MOD_MAX; i++) {
			mod_checkboxes[i]->set_pressed(false);
		}
```
(commented https://github.com/godotengine/godot/commit/95853006c582481c8c6131553f234271eece1298#r135201375)

This appears to work for me: https://github.com/godotengine/godot-proposals/assets/704698/39c9156c-8bfa-42d7-a333-4a509b43543d

**Reply by Calinou** [reply]
I've opened a PR: https://github.com/godotengine/godot/pull/86280

Thanks for the help :slightly_smiling_face:

**Reply by nrichards** [reply]
Thanks for taking my feedback seriously. Cheers!
